travis-ci/vmtest/run.sh

```diff
@@ -457,7 +458,13 @@ qemu-system-x86_64 -nodefaults -display none -serial mon:stdio -no-reboot \
 	-kernel "$vmlinuz" -append "root=/dev/vda rw console=ttyS0,115200 kernel.panic=-1 $APPEND"
 
 sudo mount -o loop "$IMG" "$mnt"
-if exitstatus="$(cat "$mnt/exitstatus" 2>/dev/null)"; then
+
+while read l; do
+	test_results[${l%:*}]=${l#*:}
```

Comment:
it would be nice to preserve the order of execution, I think this bash's map actually loses it. Is there any way to fix that?

Reply:
There's always a way, I guess, but that will probably require a bit more processing. We may have to drop the associative array and find something else, to store both the result and the order of execution.

Reply:
Yeah, given we have sequential file, we can do just that, right? It could be simpler to have "test_name 0" format to do readline later.

Reply:
Yep, we'll just need to deal with the bpftool results that we get first but that isn't written to the file. I'll sort this out.

Reply:
Why not write bpftool's results to the same file in the same format?..

Reply:
I can write to the file, I think, since we access it (read it) from outside the VM. I was a bit wary of that because the setup commands could have erased bpftool's results; but that's probably too much precaution. It also makes it slightly more difficult to know if the setup command has written its result (not an issue in the CI I think, this would be in case we run the script independently with a custom `--setup-cmd`), but it's just a matter of grepping out for `bpftool`.

I updated the PR to try and write all to the `/exitstatus` file, let's see how it goes.